services/hermes_skyportal_sync/hermes_skyportal_sync.py

```
@@ -0,0 +1,693 @@
+            log("Service interrupted")
+        finally:
+            self._cleanup()
+
+    def _process_kafka_message(self, msg) -> None:
+        """Process a single Kafka message with database session."""
+        try:
+            ts = msg.timestamp()[1]
+            if ts > 0:
+                msg_timestamp = datetime.fromtimestamp(ts / 1000, tz=timezone.utc)
+                log("=" * 60)
+                log(f"Message timestamp: {msg_timestamp.isoformat()}")
+
+            payload = msg.value()
+            if not payload:
+                log("Empty payload received")
+                return
+
+            try:
+                data = json.loads(payload)
+            except json.JSONDecodeError as e:
+                log(f"JSON parse error: {e}")
+                log(f"Raw payload: {payload[:200]}...")
+                return
+
+            session_context_id.set(uuid.uuid4().hex)
+            with DBSession() as session:
+                try:
+                    self.processor.process_message(session, data)
+                except Exception as e:
+                    log(f"Error processing message: {e}")
+                    traceback.print_exc()
+                    session.rollback()
+
+        except Exception as e:
+            log(f"Error in Kafka message handler: {e}")
+            traceback.print_exc()
+
+    def _cleanup(self) -> None:
+        """Clean up resources."""
+        if self.consumer:
+            self.consumer.close()
+            log("Hermes sync service stopped")
+
+
+@check_loaded(logger=log)
+def service(*args, **kwargs):
+    """Main service entry point with app loading check."""
+    hermes_sync_cfg = cfg.get("app.hermes.sync", {})
+
+    # Kafka/SCiMMA configuration
+    username = hermes_sync_cfg.get("scimma_username")
+    password = hermes_sync_cfg.get("scimma_password")
+    server_url = hermes_sync_cfg.get("kafka_server", "kafka.scimma.org")
+
+    topic = cfg.get("app.hermes.topic", "skyportal.skyportal_test")
+
```

We don't want to hard code any topic name. You should raise an error if no topic is available.